src/services/Queries.js

```diff
@@ -408,5 +408,78 @@ class Queries {
       return error;
     }
   }
+
+  /** Function to find a user with a manager role
+   *
+   * This service get a spacific trip that corresponds to user id and accomodation id
+   * @param {object} table table to be searching from
+   * @param {Object} userid user id
+   * @param {Object} accommodationid accomodation id
+   * @returns {object} data of the trip found that corresponds to that user and accomodation id
+   */
+  static async findIfAccomodationBooked(table, userid, accommodationid) {
+    try {
+      const tripData = table.findOne({
+        where: {
+          [Op.and]: [
+            { userid: { [Op.eq]: userid } },
+            { accommodationid: { [Op.eq]: accommodationid } }
+          ]
+        }
+      });
+      return tripData;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+ * checking if the accommodation exist
+ * @param {object} table table to be searching from
+ * @param {string} accomodationId rate data.
+ * @returns {array} data the data to be returned.
+ */
+  static async checkAccommodationById(table, accomodationId) {
+    const data = await table.findAll({
+      where: {
+        id: accomodationId
+      }
+    });
+    return data;
+  }
+
+  /**
+ * checking if the accommodation exist
+ * @param {object} table table to be searching from
+ * @param {string} accomodationId rate data.
+ * @returns {array} data the data to be returned.
+ */
+  static async checkAccommodationExistance(table, accomodationId) {
```

Comment:
@mujohn26 This function checkAccommodationExistance is the same as "checkAccommodationById" on line 442.

Reply:
I changed the function name.